Description: [PATCH] Use the window scale-factor for popup hiding
 budgie popovers need to handle window scaling.
Author: David Mohammed <fossfreedom@ubuntu.com>
Last-Update: 2019-11-18
Forwarded: https://github.com/solus-project/budgie-desktop/pull/1906

---
 src/plugin/popover.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/plugin/popover.c b/src/plugin/popover.c
index 0d1cf1ce..4cee10fc 100644
--- a/src/plugin/popover.c
+++ b/src/plugin/popover.c
@@ -903,7 +903,12 @@ static gboolean budgie_popover_button_press(GtkWidget *widget, GdkEventButton *b
         gint root_y = (gint)button->y_root;
 
         /* Inside our window? Continue as normal. */
-        if ((root_x >= x && root_x <= x + w) && (root_y >= y && root_y <= y + h)) {
+        gint scale_factor = gtk_widget_get_scale_factor(widget);
+
+        if (((root_x*scale_factor) >= x &&
+             (root_x*scale_factor) <= x + (w*scale_factor)) &&
+            ((root_y*scale_factor) >= y &&
+             (root_y*scale_factor) <= y + (h*scale_factor))) {
                 return GDK_EVENT_PROPAGATE;
         }
 
-- 
2.17.1

